src/test/java/me/karakelley/http/HttpServerTest.java

```diff
@@ -277,7 +277,37 @@ void testDisplaysFilesAtRoot() {
         e.printStackTrace();
       }
       assertTrue(response.contains("Content-Type: text/html"));
-      assertTrue(response.contains("<p>"+fileOne.getFileName()+"</p>"));
+      assertTrue(response.contains("<p><a href=\"/test1.txt\">test1.txt</a></p>"));
+    });
+  }
+
+  @Test
+  void testReturnsFileContentsRequested() {
+    TempFilesHelper.withTempDirectory(directory -> {
+      Path fileOne = TempFilesHelper.createTempFile(directory, "/test1");
+      writeToFile("Hello World", fileOne);
+      ServerConfiguration config = new ServerConfiguration();
+      config.setController(new Application(PublicDirectory.create(directory.toString(), new FileFinderCache(new RealFileFinder()))));
```

Comment:
I'm a little confused about why the argument to `setController` is an `Application`...

Reply:
It doesn't have to be `Application` but since these are my end-to-end tests and `Application` is what I initialize my program with it made sense in order to test it end-to-end.

Reply:
Ahh, I didn't see that `Application` implements the `Controller` interface. Let's talk about this outside the context of this PR.